### PR TITLE
720 changes

### PR DESCRIPTION
**Pull request by pankaj-jangir**

### CODE REVIEW

File: samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerAdsActivity.java [hunk 1]
```diff
@@ -68,42 +71,60 @@ private void setupBannerAd() {
...
             @Override
-            public void onUserLeftApplication(InMobiBanner inMobiBanner) {
-                Log.d(TAG, "onUserLeftApplication");
+            public void onRewardsUnlocked(InMobiBanner inMobiBanner, Map<Object, Object> map) {
+                super.onRewardsUnlocked(inMobiBanner, map);
+                Log.d(TAG, "onRewardsUnlocked");
+            }
+
+            @Override
+            public void onRequestPayloadCreated(byte[] bytes) {
```
Comment (davendarojha) [inline samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerAdsActivity.java]: No need to implement this method.

File: samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerAdsActivity.java [hunk 2]
```diff
@@ -68,42 +71,60 @@ private void setupBannerAd() {
...
+                super.onRequestPayloadCreated(bytes);
+                Log.d(TAG, "onRequestPayloadCreated");
             }
 
             @Override
-            public void onAdRewardActionCompleted(InMobiBanner inMobiBanner, Map<Object, Object> map) {
-                Log.d(TAG, "onAdRewardActionCompleted");
+            public void onRequestPayloadCreationFailed(InMobiAdRequestStatus inMobiAdRequestStatus) {
```
Comment (davendarojha) [inline samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerAdsActivity.java]: No need to implement this method.

File: samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerPrefetchActivity.java [hunk 3]
```diff
@@ -101,44 +103,62 @@ private void setupBannerAd() {
...
             @Override
-            public void onUserLeftApplication(InMobiBanner inMobiBanner) {
-                Log.d(TAG, "onUserLeftApplication");
+            public void onRewardsUnlocked(InMobiBanner inMobiBanner, Map<Object, Object> map) {
+                super.onRewardsUnlocked(inMobiBanner, map);
+                Log.d(TAG, "onRewardsUnlocked");
+            }
+
+            @Override
+            public void onRequestPayloadCreated(byte[] bytes) {
```
Comment (davendarojha) [inline samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerPrefetchActivity.java]: No need to implement this method.

File: samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerPrefetchActivity.java [hunk 4]
```diff
@@ -101,44 +103,62 @@ private void setupBannerAd() {
...
+                super.onRequestPayloadCreated(bytes);
+                Log.d(TAG, "onRequestPayloadCreated");
             }
 
             @Override
-            public void onAdRewardActionCompleted(InMobiBanner inMobiBanner, Map<Object, Object> map) {
-                Log.d(TAG, "onAdRewardActionCompleted");
+            public void onRequestPayloadCreationFailed(InMobiAdRequestStatus inMobiAdRequestStatus) {
```
Comment (davendarojha) [inline samples/bannerSample/src/main/java/com/inmobi/banner/sample/BannerPrefetchActivity.java]: No need to implement this method.

File: samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java [hunk 5]
```diff
@@ -127,39 +118,71 @@ public void onAdLoadSucceeded(InMobiInterstitial inMobiInterstitial) {
...
                     }
+
+                    @Override
+                    public void onRewardsUnlocked(InMobiInterstitial inMobiInterstitial, Map<Object, Object> map) {
+                        super.onRewardsUnlocked(inMobiInterstitial, map);
+                        Log.d(TAG, "onRewardsUnlocked " + map.size());
+                    }
+
+                    @Override
+                    public void onRequestPayloadCreated(byte[] bytes) {
```
Comment (davendarojha) [inline samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java]: No need to implement this method.

File: samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java [hunk 6]
```diff
@@ -127,39 +118,71 @@ public void onAdLoadSucceeded(InMobiInterstitial inMobiInterstitial) {
...
+                        super.onRequestPayloadCreated(bytes);
+                        Log.d(TAG, "onRequestPayloadCreated " + bytes);
+                    }
+
+                    @Override
+                    public void onRequestPayloadCreationFailed(InMobiAdRequestStatus inMobiAdRequestStatus) {
```
Comment (davendarojha) [inline samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java]: No need to implement this method.

File: samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java [hunk 7]
```diff
@@ -195,40 +208,74 @@ public void onAdLoadSucceeded(InMobiInterstitial inMobiInterstitial) {
...
             }
+
+            @Override
+            public void onRewardsUnlocked(InMobiInterstitial inMobiInterstitial, Map<Object, Object> map) {
+                super.onRewardsUnlocked(inMobiInterstitial, map);
+                Log.d(TAG, "onRewardsUnlocked " + map.size());
+            }
+
+            @Override
+            public void onRequestPayloadCreated(byte[] bytes) {
```
Comment (davendarojha) [inline samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java]: No need to implement this method.

File: samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java [hunk 8]
```diff
@@ -195,40 +208,74 @@ public void onAdLoadSucceeded(InMobiInterstitial inMobiInterstitial) {
...
+                super.onRequestPayloadCreated(bytes);
+                Log.d(TAG, "onRequestPayloadCreated " + bytes);
+            }
+
+            @Override
+            public void onRequestPayloadCreationFailed(InMobiAdRequestStatus inMobiAdRequestStatus) {
```
Comment (davendarojha) [inline samples/interstitialSample/src/main/java/com/inmobi/interstitial/sample/InterstitialAdsActivity.java]: No need to implement this method.

File: samples/nativeSample/src/main/java/com/inmobi/nativead/sample/listview/ListViewFeedFragment.java [hunk 9]
```diff
@@ -86,6 +87,7 @@ private void createStrands() {
         for (int position : mAdPositions) {
             final InMobiNative nativeStrand = new InMobiNative(getActivity(),
                     PlacementId.YOUR_PLACEMENT_ID_HERE, new StrandAdListener(position));
+
```
Comment (davendarojha) [inline samples/nativeSample/src/main/java/com/inmobi/nativead/sample/listview/ListViewFeedFragment.java]: Please provide example for Video events.

File: samples/nativeSample/src/main/java/com/inmobi/nativead/sample/recyclerview/RecyclerFeedFragment.java [hunk 10]
```diff
@@ -147,7 +148,7 @@ private boolean canViewScrollUp(RecyclerView recyclerView) {
         }
     }
 
-    private final class StrandAdListener implements InMobiNative.NativeAdListener {
+    private final class StrandAdListener extends NativeAdEventListener {
```
Comment (davendarojha) [inline samples/nativeSample/src/main/java/com/inmobi/nativead/sample/recyclerview/RecyclerFeedFragment.java]: Please provide example for Video events.

File: samples/nativeSample/src/main/java/com/inmobi/nativead/sample/singlestrand/SingleNativeAdFragment.java [hunk 11]
```diff
@@ -81,7 +81,7 @@ public void onConfigurationChanged(Configuration newConfig) {
     }
 
     private void createStrands() {
-        mInMobiNative = new InMobiNative(getActivity(), PlacementId.YOUR_PLACEMENT_ID_HERE, this);
+        mInMobiNative = new InMobiNative(getActivity(), PlacementId.YOUR_PLACEMENT_ID_HERE, new StrandAdListener());
```
Comment (davendarojha) [inline samples/nativeSample/src/main/java/com/inmobi/nativead/sample/singlestrand/SingleNativeAdFragment.java]: Please provide example for Video events.

**Review by davendarojha — CHANGES_REQUESTED** [review]

**Comment by davendarojha** [comment]
Please update all files under sdk folder. @pankaj-jangir

**Review by davendarojha — APPROVED** [review]

**Review by kvaddamani — APPROVED** [review]
LGTM

**Pull request merged**